lib/src/components/navigation_rail/navigation_rail.dart

```diff
@@ -0,0 +1,225 @@
+  /// ```
+  final EdgeInsets itemSpacing;
+
+  /// The padding within an item in [ZetaNavigationRail].
+  /// Default is:
+  /// ```
+  /// const EdgeInsets.symmetric(
+  ///   horizontal: ZetaSpacing.xs,
+  ///   vertical: ZetaSpacing.s,
+  /// ),
+  /// ```
+  final EdgeInsets? itemPadding;
+
+  /// Determines if words in items' labels should be wrapped on separate lines.
+  /// Default is `true`.
+  final bool? wordWrap;
+
+  @override
+  State<ZetaNavigationRail> createState() => _ZetaNavigationRailState();
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty<bool>('rounded', rounded))
+      ..add(IntProperty('selectedIndex', selectedIndex))
+      ..add(ObjectFlagProperty<void Function(int p1)?>.has('onSelect', onSelect))
+      ..add(IterableProperty<ZetaNavigationRailItem>('items', items))
+      ..add(DiagnosticsProperty<EdgeInsets>('margin', margin))
+      ..add(DiagnosticsProperty<EdgeInsets>('itemSpacing', itemSpacing))
+      ..add(DiagnosticsProperty<EdgeInsets?>('itemPadding', itemPadding))
+      ..add(DiagnosticsProperty<bool>('wordWrap', wordWrap));
+  }
+}
+
+class _ZetaNavigationRailState extends State<ZetaNavigationRail> {
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: widget.margin,
+      child: IntrinsicWidth(
+        child: Column(
+          children: [
+            for (int i = 0; i < widget.items.length; i++)
+              Row(
+                children: [
+                  Expanded(
+                    child: Padding(
+                      padding: widget.itemSpacing,
+                      child: _ZetaNavigationRailItemContent(
+                        label: widget.items[i].label,
+                        icon: widget.items[i].icon,
+                        selected: widget.selectedIndex == i,
+                        disabled: widget.items[i].disabled,
+                        onTap: () => widget.onSelect?.call(i),
+                        rounded: widget.rounded,
+                        padding: widget.itemPadding,
+                        wordWrap: widget.wordWrap,
+                      ),
+                    ),
+                  ),
+                ],
+              ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class _ZetaNavigationRailItemContent extends StatelessWidget {
+  const _ZetaNavigationRailItemContent({
+    required this.label,
+    this.icon,
+    this.selected = false,
+    this.disabled = false,
+    this.onTap,
+    this.rounded = true,
+    this.padding,
+    this.wordWrap,
+  });
+
+  final String label;
+  final Widget? icon;
+  final bool selected;
+  final bool disabled;
+  final VoidCallback? onTap;
+  final bool rounded;
+  final EdgeInsets? padding;
+  final bool? wordWrap;
+
+  @override
+  Widget build(BuildContext context) {
+    final zeta = Zeta.of(context);
+    return GestureDetector(
```

Comment:
could we wrap this container in 2 widgets:
* MouseRegion - so that we can use the click cursor when the component is used on non-mobile platforms
* SelectionContainer.disabled - so that text can not be selected